### PR TITLE
MAINT: Fix pypy3 on CircleCI

### LINKED ISSUE #9691: CircleCI tests failing on pypy3
The four most recent PRs (#9684 #9690 #9684 #9685) all have the following failure from CircleCI:

```
#!/bin/bash -eo pipefail
export CCACHE_COMPRESS=1
export PATH=/usr/lib/ccache:$PATH
# Limit parallelism for Cythonization to 4 processes, to
# avoid exceeding CircleCI memory limits
export SCIPY_NUM_CYTHONIZE_JOBS=4
export NPY_NUM_BUILD_JOBS=`pypy3 -c 'import multiprocessing as mp; print(mp.cpu_count())'`
# Less aggressive optimization flags for faster compilation
OPT="-O1" FOPT="-O1" pypy3 setup.py build
Traceback (most recent call last):
  File "setup.py", line 488, in <module>
    setup_package()
  File "setup.py", line 420, in setup_package
    import numpy
  File "/usr/local/site-packages/numpy/__init__.py", line 142, in <module>
    from . import core
  File "/usr/local/site-packages/numpy/core/__init__.py", line 16, in <module>
    from . import multiarray
  File "/usr/local/site-packages/numpy/core/multiarray.py", line 44, in <module>
    arange.__module__ = 'numpy'
AttributeError: readonly attribute '__module__'
Exited with code 1
```

See https://github.com/numpy/numpy/issues/12740.  Apparently the latest pypy release has a bug.  (so this issue may very well resolve itself soon enough)

### PR DESCRIPTION
**Pull request by larsoner**
Blindly trying the fix in https://github.com/numpy/numpy/pull/12594

Closes #9691.

### CODE REVIEW

**Comment by ilayn** [comment]
All green, thank you for the emergency fix @larsoner

File: .circleci/config.yml [hunk 1]
```diff
@@ -95,40 +95,56 @@ jobs:
...
-            export PATH=/usr/lib/ccache:$PATH
             pypy3 -mpip install --upgrade pip setuptools wheel
             pypy3 -mpip install Cython>=0.28.5
             pypy3 -mpip install numpy>=1.15.0
             pypy3 -mpip install --no-build-isolation --extra-index https://antocuni.github.io/pypy-wheels/ubuntu pytest pytest-xdist Tempita mpmath
       - run:
-          name: build
+          name: build SciPy
           command: |
-            export CCACHE_COMPRESS=1
```
Comment (pv) [inline .circleci/config.yml]: Why was this removed?

**Comment by mdhaber** [comment]
@larsoner 
I got the issue reported in #9691 in #9676 an hour ago : /
```#!/bin/bash -eo pipefail
export CCACHE_COMPRESS=1
export PATH=/usr/lib/ccache:$PATH
# Limit parallelism for Cythonization to 4 processes, to
# avoid exceeding CircleCI memory limits
export SCIPY_NUM_CYTHONIZE_JOBS=4
export NPY_NUM_BUILD_JOBS=`pypy3 -c 'import multiprocessing as mp; print(mp.cpu_count())'`
# Less aggressive optimization flags for faster compilation
OPT="-O1" FOPT="-O1" pypy3 setup.py build
Traceback (most recent call last):
  File "setup.py", line 488, in <module>
    setup_package()
  File "setup.py", line 420, in setup_package
    import numpy
  File "/usr/local/site-packages/numpy/__init__.py", line 142, in <module>
    from . import core
  File "/usr/local/site-packages/numpy/core/__init__.py", line 16, in <module>
    from . import multiarray
  File "/usr/local/site-packages/numpy/core/multiarray.py", line 44, in <module>
    arange.__module__ = 'numpy'
AttributeError: readonly attribute '__module__'
Exited with code 1
```

**Comment by ilayn** [comment]
I don't understand why these don't fail all at once but one after the other as we fix them.

**Comment by larsoner** [comment]
Did you rebase?

**Comment by larsoner** [comment]
(Travis builds a merged version of each PR, Circle builds just the branch IIRC, unless you add custom code to the config to do a merge)

**Comment by mdhaber** [comment]
No, thanks. I'll try that.
